### PR TITLE
Splitting profiling code to MiddlewareProfiler

### PR DESCRIPTION
**Pull request by fidian**
This made it much easier to test.

### CODE REVIEW

File: lib/middleware-profiler.js [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+    toString() {
+        var lines;
+
+        lines = [];
+        Object.keys(this.profiles).forEach((name) => {
+            lines.push([
+                name,
+                this.profiles[name].hits + " hits",
+                Math.round(this.profiles[name].elapsed * 1000) + " ms",
+                Math.round(1000 * this.profiles[name].elapsed / this.profiles[name].hits) + " avg"
```
Comment (AbsentSemicolon) [inline lib/middleware-profiler.js]: Would it be good to make sure we don't divide by 0? Saw an example of NaN in the test.

Comment (fidian) [inline lib/middleware-profiler.js]: I don't care about NaN.  This is pretty quick and dirty profiling, just enough to prove what middleware is causing slowdowns.

Comment (AbsentSemicolon) [inline lib/middleware-profiler.js]: Sounds good since we need this to be as quick as can be.

**Comment by AbsentSemicolon** [comment]
Looks good, nice to have it separate for easier profiling and testing.

**Pull request merged**